File: !1.ZH_ora_gyak/feladat1/Hotel.java
import java.util.ArrayList;
import java.util.List;

public class Hotel {
    // A private az nagyon fontos!
    // A get függvények nagyon fontosak ilyenkor!
    private String hotelName;
    private int rooms;
    private int avaliableRooms;
    private int bookedRooms;
    private List<String> guests;

    public Hotel(String hotelName, int rooms) {
        this.hotelName = hotelName;
        this.rooms = rooms;
        this.avaliableRooms = rooms;
        this.bookedRooms = 0;
        this.guests = new ArrayList<>();
    }

    public void bookedRooms(String guestName) {
        if (this.avaliableRooms > 0) {
            this.guests.add(guestName);
            this.avaliableRooms--;
            this.bookedRooms++;
        } else {
            System.out.println("Hiba: Nincs elérhető szoba.");
        }
    }

    public void checkoutRoom(String guestName) {
        if (this.guests.contains(guestName)) {
            this.guests.remove(guestName);
            this.avaliableRooms++;
            this.bookedRooms--;
        } else {
            System.out.println("Hiba: A vendég nem található.");
        }
    }

    public boolean isRoomAvaliable() {
        return this.avaliableRooms > 0;
    }

    public String getHotelName() {
        return this.hotelName;
    }

    public int getRooms() {
        return this.rooms;
    }

    public int getAvaliableRooms() {
        return this.avaliableRooms;
    }

    public int getBookedRooms() {
        return this.bookedRooms;
    }

    public List<String> getGuests() {
        return this.guests;
    }

    @Override
    public String toString() {
        return String.format("""
                Hotel adatai:
                - Hotel neve: %s
                - Szobák száma: %d
                - Szabad szábák száma: %d
                - Lefoglalt szobák száma: %d
                - Vendéglista: %s""",
                this.hotelName,
                this.rooms,
                this.avaliableRooms,
                this.bookedRooms,
                this.guests.isEmpty() ? "Jelengleg nincsenek vendégek a szállodában." : String.join(", ", this.guests));
    }
}

File: !2.ZH_gyak/allomanyok/feladat2/Hotel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace feladat2
{
    internal class Hotel
    {
        public string hotelName { get; set; }
        public int rooms { get; set; }
        public int availableRooms { get; set; }

        public int bookedRooms { get; set; }

        public List<string> guests { get; set; }

        public Hotel(string hotelName, int rooms)
        {
            this.hotelName = hotelName;
            this.rooms = rooms;
            availableRooms = rooms;
            bookedRooms = 0;
            guests = new List<string>();
        }

        public void bookRoom(string name)
        {
            if(availableRooms > 0)
            {
                availableRooms--;
                bookedRooms++;
                guests.Add(name);
            }
            else
            {
                Console.WriteLine("Hiba: Nem lehet lefoglalni szobát!");
            }
        }

        public void checkoutRoom(string name)
        {
            if(guests.Contains(name))
            {
                availableRooms++;
                bookedRooms--;
                guests.Remove(name);
            }
            else
            {
                Console.WriteLine("Hiba: Nem lehet kiléptetni!");
            }
        }

        public bool isRoomAvailable()
        {
            return availableRooms > 0;
        }

        public bool IsEmpty(List<string> guests)
        {
            return guests.Count == 0;
        }

        public override string ToString()
        {
            //string s = "";
            //if (IsEmpty(guests))
            //{
            //    s = "Nincsenek vendégek!";
            //}
            //else
            //{
            //    s = String.Join(", ", guests);
            //}

            guests.Sort();

            return String.Format($"""
                Szálloda neve: {hotelName}
                Összes szoba: {rooms}
                Elérhető szobák: {availableRooms}
                Foglalva: {bookedRooms}
                Vendégek: {(IsEmpty(guests) ? "Nincsenek vendégek" : String.Join(", ", guests))}
                """);
        }
    }
}
